asapdiscovery-docking/asapdiscovery/docking/scripts/run_docking_oe.py

```diff
@@ -600,26 +704,43 @@ def main():
                     break
                 except TimeoutError:
                     # This compound:xtal combination timed out
-                    print("Docking timed out for", args_list[8], flush=True)
-                    failed_runs += [args_list[8]]
+                    logger.error("Docking timed out for", args_list[9])
+                    failed_runs += [args_list[9]]
                 except pebble.ProcessExpired as e:
-                    print("Docking failed for", args_list[8], flush=True)
-                    print(f"\t{e}. Exit code {e.exitcode}", flush=True)
-                    failed_runs += [args_list[8]]
+                    logger.error("Docking failed for", args_list[9])
+                    logger.error(f"\t{e}. Exit code {e.exitcode}")
+                    failed_runs += [args_list[9]]
                 except Exception as e:
-                    print(
-                        "Docking failed for",
-                        args_list[8],
-                        "with Exception",
-                        e,
-                        flush=True,
+                    logger.error(
+                        f"Docking failed for {args_list[9]}, with Exception: {e.__class__.__name__}"
+                    )
+                    if hasattr(e, "traceback"):
+                        logger.error(e.traceback)
+                    failed_runs += [args_list[9]]
+
+                # things are going poorly, lets stop
+                if len(failed_runs) > args.max_failures:
+                    logger.critical(
+                        f"CRITICAL: Too many failures ({len(failed_runs)}/{args.max_failures}) while running docking, aborting"
                     )
-                    print(e.traceback, flush=True)
-                    failed_runs += [args_list[8]]
-            print(f"Docking failed for {len(failed_runs)} runs", flush=True)
+                    res.cancel()
+
+            logging.info(
+                f"Docking complete with {len(failed_runs)} failures, use --verbose to see which ones."
+            )
+            if args.verbose:
+                if len(failed_runs) > 0:
+                    failed_run_str = "\n".join(failed_runs)
+                    logger.error(f"Failed runs:\n{failed_run_str}\n")
```

Comment:
just for a minor readability improvement:
```suggestion
            if args.verbose:
                logging.info(f"Docking complete with {len(failed_runs)} failures.")
                if len(failed_runs) > 0:
                    failed_run_str = "\n".join(failed_runs)
                    logger.error(f"Failed runs:\n{failed_run_str}\n")
            else:
                logging.info(
                    f"Docking complete with {len(failed_runs)} failures, use --verbose to see which ones."
                )

```